src/controllers/mainController.ts

```diff
@@ -276,6 +277,14 @@ export default class MainController implements vscode.Disposable {
             this._event.on(Constants.cmdDisableActualPlan, () => {
                 this.onToggleActualPlan(false);
             });
+
+            this._context.subscriptions.push(
+                vscode.languages.registerCodeLensProvider(
+                    { language: "sql" },
```

Comment:
Does this mean CodeLens will also show up for non-TSQL queries as long as the editor is set to `sql`? This might get annoying if a user is working on non-TSQL queries with MSSQL extension installed.
IIRC we added the ability to disable intellisense for non-MSSQL editor (by detecting error #), not sure if something similar can be done with this.
https://github.com/microsoft/vscode-mssql/pull/18557

Reply:
Yeah, for now. However, we are having the same problem for the toolbar and status view where we show mssql options for non-tsql files. 

We probably need to improve that option to turn off all those things.

Reply:
makes sense, we need to "smartly" enable MSSQL features when needed. IMO this specific change in the editor can be more invasive than for example in the toolbar but I'm fine with having a separate issue to track all these as a whole.